scripts/o3de/o3de/repo_properties.py

```diff
@@ -0,0 +1,377 @@
+                       repo_name: str = None,
+                       add_gems: pathlib.Path or list = None,
+                       delete_gems: str or list = None,
+                       replace_gems: pathlib.Path or list = None,
+                       add_projects: pathlib.Path or list = None,
+                       delete_projects: str or list = None,
+                       replace_projects: pathlib.Path or list = None,
+                       add_templates: pathlib.Path or list = None,
+                       delete_templates: str or list = None,
+                       replace_templates: pathlib.Path or list = None,
+                       release_archive_path: pathlib.Path = None,
+                       download_prefix: str = None
+                       ) -> int:
+    """
+    Edits and modifies the remote repo properties for the repo.json located at 'repo_path'.
+    :param repo_path: The path to the repo.json file
+    :param repo_name: The new name for the remote repo
+    :param add_gems: Any gem paths to be added to the list
+    :param delete_gems: Any gem names to be removed from the list
+    :param replace_gems: A list of gem paths that will completely replace the current list of path
+    :add_projects: Any project paths to be added to the list
+    :delete_projects:  Any project names to be removed from the list
+    :replace_projects: A list of project paths that will completely replace the current list of path
+    :add_templates: Any template paths to be added to the list
+    :delete_templates: Any template names to be removed from the list
+    :replace_templates: A list of template paths that will completely replace the current list of path
+    :release_archive_path: Path where you want your release to be located
+    :download_prefix: The string prefix of the download uri
+    """
+    repo_json = get_repo_props(repo_path)
+
+    if not repo_json:
+        return 1
+
+    if isinstance(repo_name, str):
+        if not utils.validate_identifier(repo_name):
+            logger.error(f'Repo name must be fewer than 64 characters, contain only alphanumeric, "_" or "-"'
+                         f' characters, and start with a letter.  {repo_name}')
+            return 1
+        repo_json['repo_name'] = repo_name
+
+    if add_gems or delete_gems or replace_gems:
+        _edit_objects('gem', validation.valid_o3de_gem_json, repo_json, add_gems, delete_gems, replace_gems, release_archive_path, download_prefix)
+
+    if add_projects or delete_projects or replace_projects:
+        _edit_objects('project', validation.valid_o3de_project_json, repo_json, add_projects, delete_projects, replace_projects, release_archive_path, download_prefix)
+
+    if add_templates or delete_templates or replace_templates:
+        _edit_objects('template', validation.valid_o3de_template_json, repo_json, add_templates, delete_templates, replace_templates, release_archive_path, download_prefix)
+
```

Comment:
In the future, this is where we want to validate the modified `repo.json` file and backup the existing `repo.json` file before we write the updates to disk.